lib/cluster.js

```diff
@@ -0,0 +1,151 @@
+'use strict';
+
+/**
+ * In cluster masters, extends Registry class with a `clusterMetrics` method,
+ * which returns aggregated metrics for all workers.
+ *
+ * In cluster workers, listens for and responds to requests for metrics by the
+ * cluster master.
+ */
+
+const cluster = require('cluster');
+const Registry = require('./registry');
+const AGGREGATORS = require('./metricAggregators').AGGREGATORS;
+
+const GET_METRICS_REQ = 'prom-client:getMetricsReq';
+const GET_METRICS_RES = 'prom-client:getMetricsRes';
+
+// Default metrics that don't use summation:
+const DEFAULT_METRIC_AGGREGATORS = {
+	nodejs_version_info: AGGREGATORS.FIRST,
+	process_start_time_seconds: AGGREGATORS.OMIT,
+	nodejs_eventloop_lag_seconds: AGGREGATORS.AVERAGE
```

Comment:
feels sorta icky to hard-code these. what about custom metrics? Can you filter out all non-sum metrics by checking the metric's type at aggregation-time?